code/game/turfs/open/floor/catwalk_plating.dm

```diff
@@ -24,11 +25,11 @@
 
 /turf/open/floor/catwalk_floor/update_overlays()
 	. = ..()
-	var/static/image/catwalk_overlay
+	var/image/catwalk_overlay	///Per-tile instead of static, so we can have multiple different types of catwalk
 	if(isnull(catwalk_overlay))
 		catwalk_overlay = new()
 		catwalk_overlay.icon = icon
-		catwalk_overlay.icon_state = "catwalk_above"
+		catwalk_overlay.icon_state = "[above_state]"
```

Comment:
This does not spark joy
Please cache this

Reply:
Each time update_overlays is called, what we'd have now would need to generate a new catwalk overlay. This is a waste of raw time because it's the same overlay over and over again, and it's also a waste of memory because we may have say 10000 platings all with unique images.

Static vars inside a proc like that one persist through calls, so there'd only ever be 1 catwalk overlay.
What you've got now changes that slightly, since now you need a new catwalk overlay per above state.
I'd like you to cache this somehow. Maybe a global list of above_state -> catwalk overlay, then in this proc you check if the list has the overlay you want, if it does you use it and if not you generate one and add it to the list

Reply:
```
GLOBAL_LIST_INIT(catwalk_overlay_masterlist, list())	//Stores all the above_states for the different types of catwalk

/turf/open/floor/catwalk_floor/update_overlays()
	. = ..()
	if(covered)
		if(above_state in GLOB.catwalk_overlay_masterlist)
			. += GLOB.catwalk_overlay_masterlist[above_state]
		else
			var/image/catwalk_overlay = new()
			catwalk_overlay.icon = icon
			catwalk_overlay.icon_state = above_state
			catwalk_overlay.plane = GAME_PLANE
			catwalk_overlay.layer = CATWALK_LAYER
			GLOB.catwalk_overlay_masterlist[above_state] = catwalk_overlay
			. += catwalk_overlay```
new code look good?

Reply:
wow i need to learn formatting better
@LemonInTheDark

Reply:
i'm gonna actually mark this as resolved, if the new code has an issue re-review that section pls